osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -68,11 +70,35 @@ public SampleChannel GetSample(ISampleInfo sampleInfo)
 
         public IBindable<TValue> GetConfig<TLookup, TValue>(TLookup lookup)
         {
-            if (AllowConfigurationLookup && skin != null)
+            if (skin != null)
             {
-                var bindable = skin.GetConfig<TLookup, TValue>(lookup);
-                if (bindable != null)
-                    return bindable;
+                switch (lookup)
+                {
+                    // todo: the GlobalSkinColours switch is pulled from LegacySkin and should not exist.
+                    // will likely change based on how databased storage of skin configuration goes.
```

Comment:
Not sure what this comment is trying to convey - do you mind elaborating on why this is todo/temporary/should not exist? Although I do agree readability is not superb with these nested switches.

Reply:
It was present in the code where I found the GlobalSkinColours related logic from - osu.Game.Skinning.DefaultSkin.cs. I copied the comment as well since I wasn't entirely sure what it meant and wanted to make sure the same information was related to the updated code here.